emcc.py

```diff
@@ -3241,17 +3241,23 @@ def phase_embind_emit_tsd(options, in_wasm, wasm_target, memfile, js_syms):
   # Required function to trigger TS generation.
   settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$callRuntimeCallbacks']
   settings.EXPORT_ES6 = False
+  # Disable minify since the binaryen pass has not been run yet to change the
+  # import names.
+  settings.MINIFY_WASM_IMPORTED_MODULES = False
   setup_environment_settings()
   # Replace embind with the TypeScript generation version.
   embind_index = settings.JS_LIBRARIES.index('embind/embind.js')
   settings.JS_LIBRARIES[embind_index] = 'embind/embind_ts.js'
 
   outfile_js = os.path.join(os.path.dirname(wasm_target), 'tsgen_a.out.js')
-  emscripten.run(in_wasm, wasm_target, outfile_js, memfile, js_syms)
+  # The Wasm outfile may be modified by emscripten.run, so use a temporary file.
+  outfile_wasm = os.path.join(os.path.dirname(wasm_target), 'tsgen_a.out.wasm')
+  emscripten.run(in_wasm, outfile_wasm, outfile_js, memfile, js_syms)
```

Comment:
I think we should really be using temp files here.. then we wouldn't need to do delete them after.

One way to do that is using `in_temp(..)`

Reply:
Yeah, I think how we had it before we couldn't do that, but now that we control where both the wasm and js are we can.